pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -298,26 +311,28 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		c.versionRecorder.SetVersion(operandName, c.targetOperandVersion)
 	}
 
-	updateGenerationFn := func(newStatus *operatorv1.OperatorStatus) error {
+	// set updateGenerationFn so that it is invoked in defer
+	updateGenerationFn = func(newStatus *operatorv1.OperatorStatus) error {
 		resourcemerge.SetDeploymentGeneration(&newStatus.Generations, workload)
 		return nil
 	}
 
-	if _, _, updateError := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(deploymentAvailableCondition),
-		v1helpers.UpdateConditionFn(deploymentDegradedCondition),
-		v1helpers.UpdateConditionFn(deploymentProgressingCondition),
-		v1helpers.UpdateConditionFn(workloadDegradedCondition),
-		updateGenerationFn); updateError != nil {
-		return updateError
-	}
-
 	if len(errs) > 0 {
 		return kerrors.NewAggregate(errs)
 	}
 	return nil
 }
 
+func isUpdatingTooLong(operatorClient v1helpers.OperatorClient, progressingConditionType string) (bool, error) {
+	_, status, _, err := operatorClient.GetOperatorState()
```

Comment:
could we pass and use the spec from the `sync` method to avoid one extra call to API?

Reply:
it's usually cached, but yeah, I did not notice the call up in the sync, that would be better :+1: